bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Widget.java

```diff
@@ -2637,17 +2637,21 @@ void notifyDisposalTracker() {
 
 
 /**
- * The current DPI zoom level the widget is scaled for
+ * The current native zoom level of the monitor, the widget is scaled with respect to
  */
-int getZoom() {
-	return zoom;
+int getNativeZoom() {
+	return nativeZoom;
+}
+
+void setNativeZoom(int zoom) {
+	this.nativeZoom = zoom;
 }
 
-void setZoom(int zoom) {
-	this.zoom = zoom;
+int getZoom() {
+	return DPIUtil.getZoomForAutoscaleProperty(getNativeZoom());
```

Comment:
Even though it is generally good practice to access fields via accessors, I think in this case we could directly access `nativeZoom` instead of using its accessor. Since this method will be called really often, having direct access to the field and not requiring the JIT to inline the access may be reasonable (but could also be considered a premature optimization).